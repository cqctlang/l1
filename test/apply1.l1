@define fn(a ...)
{
	printf("%a -> ", a);
	return length(a);
}

printf("%a\n", apply(fn, []));
printf("%a\n", apply(fn, [1]));
printf("%a\n", apply(fn, [1, 2]));
printf("%a\n", apply(fn, [1, 2, 3]));

printf("%a\n", apply(fn, 1, []));
printf("%a\n", apply(fn, 1, [2]));
printf("%a\n", apply(fn, 1, [2, 3]));
printf("%a\n", apply(fn, 1, 2, [3]));
printf("%a\n", apply(fn, 1, 2, [3, 4]));
printf("%a\n", apply(fn, 1, 2, [3, 4, 5]));
