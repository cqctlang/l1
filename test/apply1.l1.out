entry: var params, 2 locals, 0 temps
fn: var params, 1 locals, 0 temps
[ ] -> 0
[ 1 ] -> 1
[ 1, 2 ] -> 2
[ 1, 2, 3 ] -> 3
[ 1 ] -> 1
[ 1, 2 ] -> 2
[ 1, 2, 3 ] -> 3
[ 1, 2, 3 ] -> 3
[ 1, 2, 3, 4 ] -> 4
[ 1, 2, 3, 4, 5 ] -> 5
